beir/generation/models/auto_model.py

```diff
@@ -6,16 +6,17 @@
 
 logger = logging.getLogger(__name__)
 
+
 class QGenModel:
-    def __init__(self, model_path: str, gen_prefix: str = "", use_fast: bool = True, **kwargs):
+    def __init__(self, model_path: str, gen_prefix: str = "", use_fast: bool = True, device: str = None, **kwargs):
         self.tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=use_fast)
         self.model = AutoModelForSeq2SeqLM.from_pretrained(model_path)
         self.gen_prefix = gen_prefix
-        self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        self.device = device or ('cuda' if torch.cuda.is_available() else 'cpu')
         logger.info("Use pytorch device: {}".format(self.device))
         self.model = self.model.to(self.device)
     
-    def generate(self, corpus: List[Dict[str, str]], ques_per_passage: int, top_p: int, top_k: int, max_length: int) -> List[str]:
+    def generate(self, corpus: List[Dict[str, str]], ques_per_passage: int, top_p: float, top_k: int, max_length: int) -> List[str]:
```

Comment:
@NThakur20 Note that there is a bug on `main` where `top_p` is `int` but should be `float`. This is fixed in this PR but of course if you would like to not accept this PR, this change should still be made on `main` 😄